### PR TITLE
Copybook references

### PR DESCRIPTION
**Pull request by abdrabrcom**
#107

Added support for finding copybook references

### CODE REVIEW

**Comment by zacanbrcom** [comment]
@abdrabrcom There are test failures when I tried to run the build, please review them:

```
[INFO] Running com.ca.lsp.cobol.utils.KeywordsTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.685 s - in com.ca.lsp.cobol.utils.KeywordsTest
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR] com.ca.lsp.cobol.service.MultiDocumentDefinitionTest.testVariableDefinitions(com.ca.lsp.cobol.service.MultiDocumentDefinitionTest)
[ERROR]   Run 1: MultiDocumentDefinitionTest.testVariableDefinitions:97 expected:<5> but was:<1>
[ERROR]   Run 2: MultiDocumentDefinitionTest.testVariableDefinitions:97 expected:<5> but was:<1>
[INFO] 
[ERROR] com.ca.lsp.cobol.usecases.TestCopybookWithIndirectRecursiveDependencyIsDetected.test(com.ca.lsp.cobol.usecases.TestCopybookWithIndirectRecursiveDependencyIsDetected)
[ERROR]   Run 1: TestCopybookWithIndirectRecursiveDependencyIsDetected.test:59->NegativeUseCase.test:38->NegativeUseCase.retrieveRanges:49 No diagnostics received
[ERROR]   Run 2: TestCopybookWithIndirectRecursiveDependencyIsDetected.test:59->NegativeUseCase.test:38->NegativeUseCase.retrieveRanges:49 No diagnostics received
[INFO] 
[ERROR] com.ca.lsp.cobol.usecases.TestCopybookWithRecursiveDependencyIsDetected.test(com.ca.lsp.cobol.usecases.TestCopybookWithRecursiveDependencyIsDetected)
[ERROR]   Run 1: TestCopybookWithRecursiveDependencyIsDetected.test:55->NegativeUseCase.test:38->NegativeUseCase.retrieveRanges:49 No diagnostics received
[ERROR]   Run 2: TestCopybookWithRecursiveDependencyIsDetected.test:55->NegativeUseCase.test:38->NegativeUseCase.retrieveRanges:49 No diagnostics received
[INFO] 
[ERROR] com.ca.lsp.cobol.usecases.TestMissingCopybooksReturnsError.test(com.ca.lsp.cobol.usecases.TestMissingCopybooksReturnsError)
[ERROR]   Run 1: TestMissingCopybooksReturnsError.test:49->NegativeUseCase.test:38->NegativeUseCase.retrieveRanges:49 No diagnostics received
[ERROR]   Run 2: TestMissingCopybooksReturnsError.test:49->NegativeUseCase.test:38->NegativeUseCase.retrieveRanges:49 No diagnostics received
[INFO] 
[INFO] 
[ERROR] Tests run: 232, Failures: 4, Errors: 0, Skipped: 8
```

File: com.ca.lsp.cobol/lsp-core-cobol-parser/src/main/java/com/ca/lsp/core/cobol/preprocessor/sub/document/impl/CobolSemanticParserListenerImpl.java [hunk 1]
```diff
@@ -193,7 +193,7 @@ private void defineCopybook(String copybookName, Position position) {
       return;
     }
     if (checkThisCopybookNotPresentInHierarchy(copybookName)) {
-      semanticContext.getCopybooks().define(copybookName, position);
+      semanticContext.getCopybooks().addUsage(copybookName, position);
       semanticContext.getVariables().define(new Variable("-1", copybookName), position);
```
Comment (temanbrcom) [inline com.ca.lsp.cobol/lsp-core-cobol-parser/src/main/java/com/ca/lsp/core/cobol/preprocessor/sub/document/impl/CobolSemanticParserListenerImpl.java]: It seems like the problem with the test caused by this line. I think `define()` here should be also replaced with `addUsage()`

Comment (temanbrcom) [reply com.ca.lsp.cobol/lsp-core-cobol-parser/src/main/java/com/ca/lsp/core/cobol/preprocessor/sub/document/impl/CobolSemanticParserListenerImpl.java]: Also, try to replace definitions call in `CobolVariableContext#buildVariableStructure()` with one for usages

Comment (temanbrcom) [reply com.ca.lsp.cobol/lsp-core-cobol-parser/src/main/java/com/ca/lsp/core/cobol/preprocessor/sub/document/impl/CobolSemanticParserListenerImpl.java]: It was not that, the problem was in `CobolSemanticParserImpl#processCopybooks`

File: com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/service/delegates/references/DocumentOccurrencesTest.java [hunk 2]
```diff
@@ -159,6 +175,23 @@ public void testFindDeclarationForParagraph() throws ExecutionException, Interru
...
+    assertContainsRange(locations, range(20, 12, COPY_LENGTH));
+    assertContainsRange(locations, range(22, 12, COPY_LENGTH));
+  }
+
+  @Test
+  public void testFindSingleCopybookReference() throws ExecutionException, InterruptedException {
+    List<? extends Location> locations = invokeReferencesRequest(TEST_COPYBOOK2, true);
+    assertEquals(1, locations.size());
+
+    assertContainsRange(locations, range(21, 12, COPY_LENGTH));
```
Comment (sergiuilie) [inline com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/service/delegates/references/DocumentOccurrencesTest.java]: could you please provide a negative scenario as well in which you will try to get a reference and there is none

**Comment by sergiuilie** [comment]
please solve the conflict firstly

**Pull request merged**